src/sentry_value.c

```diff
@@ -26,38 +26,47 @@
 #include "sentry_value.h"
 
 /**
- * NaN-boxing of sentry_value_t
+ * Pointer Tagging of `sentry_value_t`
  *
- * We do encode 32-bit values as well as some constants and pointers into the
- * NaN-space of 64-bit floating-point numbers, like so:
+ * We expect all of the pointers we deal with to be at least 4-byte aligned,
+ * which means we can use the least significant 2 bits for tagging.
```

Comment:
do we have any way of checking this is true at build time rather than mysteriously fail at runtime?

Reply:
We only ever use pointers to `thing_t`, which we can check has an `alignof` of `>= 4` I guess.
Other than that, both our own allocator, as well as the system allocator is pretty much guaranteed to give us 8-byte alignment I think.

Reply:
Since we both asked this, maybe write this into the comment :)